cardano-crypto-tests/bls12-381-test-vectors/test-vectors-bls12_381/src/main.rs

```diff
@@ -0,0 +1,338 @@
+#![warn(missing_docs)]
+#![doc = include_str!("../README.md")]
+#![allow(non_snake_case)]
+
+extern crate core;
+use bls12_381::{G1Affine, G1Projective, G2Affine, G2Projective, Scalar};
+use blst::min_sig::*;
+use blst::BLST_ERROR;
+use ff::Field;
+use group::Group;
+use rand_chacha::rand_core::{RngCore, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+use std::fs::File;
+use std::io::prelude::*;
+use std::ops::{Add, AddAssign, Mul, MulAssign, Neg, Sub};
+
+fn pairing_properties<R: RngCore>(mut rng: R) -> std::io::Result<()> {
+    let P = G1Affine::from(G1Projective::random(&mut rng));
+    let Q = G2Affine::from(G2Projective::random(&mut rng));
+
+    let a = Scalar::random(&mut rng);
+    let b = Scalar::random(&mut rng);
+    let mut aplusb = Scalar::clone(&a);
+    let mut atimesb = Scalar::clone(&a);
+
+    aplusb.add_assign(&b);
+    atimesb.mul_assign(&b);
+
+    let aP = G1Affine::from(P.mul(a));
+    let bQ = G2Affine::from(Q.mul(b));
+    let bP = G1Affine::from(P.mul(b));
+    let aQ = G2Affine::from(Q.mul(a));
+
+    let aplusbP = G1Affine::from(P.mul(aplusb));
+    let atimesbP = G1Affine::from(P.mul(atimesb));
+
+    let hex_aP = hex::encode(aP.to_compressed());
+    let hex_bQ = hex::encode(bQ.to_compressed());
+    let hex_bP = hex::encode(bP.to_compressed());
+    let hex_aQ = hex::encode(aQ.to_compressed());
+    let hex_aplusbP = hex::encode(aplusbP.to_compressed());
+    let hex_atimesbP = hex::encode(atimesbP.to_compressed());
+
+    let mut file = File::create("pairing_test_vectors")?;
+    file.write_all(hex_aP.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_bQ.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_bP.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_aQ.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_aplusbP.as_ref())?;
+    file.write_all(b"\n")?;
+    file.write_all(hex_atimesbP.as_ref())?;
```

Comment:
You could create a helper function to take as input a slice of hex strings. That would simplify the code of this script.